butterfly-utilities/src/main/java/com/paypal/butterfly/utilities/operations/xml/XmlXPathReplace.java

```diff
@@ -0,0 +1,184 @@
+     * the result of this transformation utility.
+     * 
+     * @param xpathExpressionString a string that compiles into a {@link javax.xml.xpath.XPathExpression}
+     *   if the expression evaluates to an empty string 'false' is returned, otherwise true
+     *
+     * @return this instance
+     */
+    public XmlXPathReplace setXPathExpression(String xpathExpressionString) {
+        checkForBlankString("XPath Expression", xpathExpressionString);
+        this.xpathExpression = checkXPathCompile(xpathExpressionString);
+        this.xpathExpressionString = xpathExpressionString;
+        return this;
+    }
+
+    private XPathExpression checkXPathCompile(String expression) throws TransformationDefinitionException {
+        XPathFactory xpathFactory = XPathFactory.newInstance();
+        XPath xpath = xpathFactory.newXPath();
+        XPathExpression expr = null;
+
+        try {
+            expr = xpath.compile(expression);
+        } catch (XPathExpressionException e) {
+            throw new TransformationDefinitionException(
+                    "XPath expression '" + expression + "' didn't compile correctly.");
+        }
+
+        return expr;
+    }
+
+    @Override
+    protected TOExecutionResult execution(File transformedAppFolder, TransformationContext transformationContext) {
+        File fileToBeChanged = getAbsoluteFile(transformedAppFolder, transformationContext);
+        TOExecutionResult result = null;
+        NodeList nodes = null;
+
+        try {
+            File readFile = getOrCreateReadFile(transformedAppFolder, transformationContext);
+            DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+            DocumentBuilder builder;
+            builder = factory.newDocumentBuilder();
+            Document doc = builder.parse(readFile);
+            nodes = (NodeList) xpathExpression.evaluate(doc, XPathConstants.NODESET);
+
+            // make the change
+            for (int idx = 0; idx < nodes.getLength(); idx++) {
+                Node node = nodes.item(idx);
+
+                if (this.textReplace) {
+                    node.setTextContent((String)replacementObject);                    
+                } else {
+                    Node newNode = doc.importNode((Element)replacementObject, true);
+                    node.getParentNode().replaceChild(newNode, node);
+                }
+            }
+
+            // save the result
+            Transformer xformer = TransformerFactory.newInstance().newTransformer();
+            xformer.transform(new DOMSource(doc), new StreamResult(fileToBeChanged));
+
+        } catch (ParserConfigurationException | SAXException | IOException e) {
+            result = TOExecutionResult.error(this,
+                    new TransformationOperationException("File content could not be parsed properly in XML format", e));
+        } catch (TransformationUtilityException e) {
```

Comment:
You can have catch `TransformationUtilityException `, `TransformerConfigurationException` and `TransformerException` in a single catch block by doing this:

``` java
} catch (TransformationUtilityException | TransformerConfigurationException | TransformerException e) {
   result = TOExecutionResult.error(this, e);
} 
```

[More details here.](https://www.journaldev.com/629/java-catch-multiple-exceptions-rethrow-exception)